File: src/didery/static/transcrypt/components/tabs.py
# ================================================== #
#                        TABS                        #
# ================================================== #
# Author: Brady Hammond                              #
# Created: 05/08/2018                                #
# Last Edited:                                       #
# Last Edited By:                                    #
# ================================================== #
#                      IMPORTS                       #
# ================================================== #

import didery.static.transcrypt.components.tabledtab as tabledtab
import didery.static.transcrypt.components.tables as tables

# ================================================== #
#                  CLASS DEFINITIONS                 #
# ================================================== #

class Errors(tabledtab.TabledTab):
    """
    Class for errors tab.
    """
    Name = "Errors"
    Icon = "i.exclamation.circle.icon"
    DataTab = "errors"
    Active = False

    def setup_table(self):
        """
        Sets up errors table.
        """
        self.table = tables.ErrorsTable()

# ================================================== #

class Relays(tabledtab.TabledTab):
    """
    Class for relays tab.
    """
    Name = "Relays"
    Icon = "i.server.icon"
    DataTab = "relays"
    Active = False

    def setup_table(self):
        """
        Sets up relays table.
        """
        self.table = tables.RelaysTable()

# ================================================== #

class Blobs(tabledtab.TabledTab):
    """
   Class for encrypted blobs tab.
    """
    Name = "Encrypted Blobs"
    Icon = "i.unlock.alternate.icon"
    DataTab = "blobs"
    Active = False

    def setup_table(self):
        """
        Sets up blob table.
        """
        self.table = tables.BlobsTable()

# ================================================== #

class History(tabledtab.TabledTab):
    """
   Class for public keys tab.
    """
    Name = "Public Keys"
    Icon = "i.key.icon"
    DataTab = "history"
    Active = True

    def setup_table(self):
        """
        Sets up history table.
        """
        self.table = tables.HistoryTable()

# ================================================== #
#                        EOF                         #
# ================================================== #

File: src/didery/static/js/components/tabs.js
let tables = require("./tables");
let tabled_tab = require("./tabledtab");

module.exports = {
    ErrorsTab: class Errors extends tabled_tab.TabledTab{
        constructor() {
            super();
            this.Name = "Errors";
            this.Icon = "i.exclamation.circle.icon";
            this.DataTab = "errors";
            this.Active = false;

            this._menu_attrs = {"data-tab": this.DataTab};
            this._tab_attrs = {"data-tab": this.DataTab};

            if (this.Active) {
                this._menu += ".active";
                this._tab += ".active";
            }

            this._detailsId = this.DataTab + "DetailsCodeBlock";
            this._copiedId = this.DataTab + "CopiedCodeBlock";
            this._copyButtonId = this.DataTab + "CopyButton";
            this._clearButtonId = this.DataTab + "ClearButton";
        }

        setup_table () {
            this.table = new tables.ErrorsTable();
        }
    },

    RelaysTab: class Relays extends tabled_tab.TabledTab{
        constructor() {
            super();
            this.Name = "Relays";
            this.Icon = "i.server.icon";
            this.DataTab = "relays";
            this.Active = false;

            this._menu_attrs = {"data-tab": this.DataTab};
            this._tab_attrs = {"data-tab": this.DataTab};

            if (this.Active) {
                this._menu += ".active";
                this._tab += ".active";
            }

            this._detailsId = this.DataTab + "DetailsCodeBlock";
            this._copiedId = this.DataTab + "CopiedCodeBlock";
            this._copyButtonId = this.DataTab + "CopyButton";
            this._clearButtonId = this.DataTab + "ClearButton";
        }

        setup_table () {
            this.table = new tables.RelaysTable();
        }
    },

    BlobsTab: class Blobs extends tabled_tab.TabledTab{
        constructor() {
            super();
            this.Name = "Encrypted Blobs";
            this.Icon = "i.unlock.alternate.icon";
            this.DataTab = "blobs";
            this.Active = false;

            this._menu_attrs = {"data-tab": this.DataTab};
            this._tab_attrs = {"data-tab": this.DataTab};

            if (this.Active) {
                this._menu += ".active";
                this._tab += ".active";
            }

            this._detailsId = this.DataTab + "DetailsCodeBlock";
            this._copiedId = this.DataTab + "CopiedCodeBlock";
            this._copyButtonId = this.DataTab + "CopyButton";
            this._clearButtonId = this.DataTab + "ClearButton";
        }

        setup_table () {
            this.table = new tables.BlobsTable();
        }
    },

    HistoryTab: class History extends tabled_tab.TabledTab{
        constructor() {
            super();
            this.Name = "Public Keys";
            this.Icon = "i.key.icon";
            this.DataTab = "history";
            this.Active = true;

            this._menu_attrs = {"data-tab": this.DataTab};
            this._tab_attrs = {"data-tab": this.DataTab};

            if (this.Active) {
                this._menu += ".active";
                this._tab += ".active";
            }

            this._detailsId = this.DataTab + "DetailsCodeBlock";
            this._copiedId = this.DataTab + "CopiedCodeBlock";
            this._copyButtonId = this.DataTab + "CopyButton";
            this._clearButtonId = this.DataTab + "ClearButton";
        }

        setup_table () {
            this.table = new tables.HistoryTable();
        }
    }
};